routes/dao_coin_exchange.go

```diff
@@ -0,0 +1,197 @@
+	TransactorPublicKeyBase58Check string `safeForLogging:"true"`
+
+	BuyingDAOCoinCreatorPublicKeyBase58Check  string `safeForLogging:"true"`
+	SellingDAOCoinCreatorPublicKeyBase58Check string `safeForLogging:"true"`
+
+	// One of these two should be populated
+	ScaledExchangeRateCoinsToSellPerCoinToBuy *uint256.Int `safeForLogging:"true"`
+	ExchangeRateCoinsToSellPerCoinToBuy       float64      `safeForLogging:"true"`
+
+	// One of these two should be populated
+	QuantityToBuyInBaseUnits *uint256.Int `safeForLogging:"true"`
+	QuantityToBuy            float64      `safeForLogging:"true"`
+}
+
+func (fes *APIServer) GetDAOCoinLimitOrders(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetDAOCoinLimitOrdersRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("GetDAOCoinLimitOrders: Problem parsing request body: %v", err),
+		)
+		return
+	}
+
+	if requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername == "" &&
+		requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername == "" {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("GetDAOCoinLimitOrders: Must provide either a "+
+				"DAOCoin1CreatorPublicKeyBase58CheckOrUsername or DAOCoin2CreatorPublicKeyBase58CheckOrUsername "+
+				"or both"),
+		)
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Problem fetching utxoView: %v", err))
+		return
+	}
+
+	coin1PKID := &lib.ZeroPKID
+	coin2PKID := &lib.ZeroPKID
+
+	coin1ProfilePublicBase58Check := ""
+	coin2ProfilePublicBase58Check := ""
+
+	if requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername != "" {
+		coin1ProfilePublicBase58Check, coin1PKID, err = fes.validateCreatorPublicKeyBase58CheckOrUsername(
+			utxoView,
+			requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername,
+		)
+		if err != nil {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprintf("GetDAOCoinLimitOrders: Invalid DAOCoin1CreatorPublicKeyBase58CheckOrUsername: %v", err),
+			)
+			return
+		}
+	}
+
+	if requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername != "" {
+		coin2ProfilePublicBase58Check, coin2PKID, err = fes.validateCreatorPublicKeyBase58CheckOrUsername(
+			utxoView,
+			requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername,
+		)
+		if err != nil {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprintf("GetDAOCoinLimitOrders: Invalid DAOCoin2CreatorPublicKeyBase58CheckOrUsername: %v", err),
+			)
+			return
+		}
+	}
+
+	adapter := utxoView.GetDbAdapter()
+
+	ordersBuyingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin1PKID, coin2PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	ordersSellingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin2PKID, coin1PKID)
```

Comment:
I think we should use `utxoView._getAllDAOCoinLimitOrdersForThisDAOCoinPair` instead of the dbAdapter so we also return results from the mempool